parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -795,6 +804,36 @@ public static Statistics toParquetStatistics(
     return formatStats;
   }
 
+  private static BoundingBox toParquetBoundingBox(org.apache.parquet.column.statistics.geometry.BoundingBox bbox) {
+    // Check if any of the required bounding box is valid.
+    if (!bbox.isValid() || bbox.isEmpty()) {
+      // According to the Thrift-generated class, these fields are marked as required and must be set explicitly.
+      // If any of them is NaN, it indicates the bounding box is invalid or uninitialized,
+      // so we return null to avoid creating a malformed BoundingBox object that would later fail serialization
+      // or validation.
+      return null;
+    }
+
+    // Now we can safely create the BoundingBox object
+    BoundingBox formatBbox = new BoundingBox();
+    formatBbox.setXmin(bbox.getXMin());
+    formatBbox.setXmax(bbox.getXMax());
+    formatBbox.setYmin(bbox.getYMin());
+    formatBbox.setYmax(bbox.getYMax());
+
+    if (bbox.isZValid()) {
```

Comment:
What if Z or M is valid but empty? Should it be omitted here? That's why I suggest to add `isXYEmpty|isZEmpty|isMEmpty` separately.